### PR TITLE
`number_of_days()` second PR bug fix

### PR DESCRIPTION
**Pull request by viettran15**
## Summary
- Fixing bugs that caused test failure

## Details
- Updating the logic within the conditional statement to correctly account for leap year
- Changing the months that contain 31 days versus 30 days

### CODE REVIEW

**Review by YLee2121 — CHANGES_REQUESTED** [review]
It doesn't pass the test case (1900, 2).

It's not a leap year for every 100 years except 400 years. For example, 1800 and 1900 are not a leap year but 2000 is a leap year.

```console
=================================== FAILURES ===================================
________________________ Test_Backend.test_num_of_days _________________________

self = <Calendar_CLI.tests.test_backed.Test_Backend object at 0x7f12f0ae10c0>

    def test_num_of_days(self):
        assert number_of_days(2023, 1) == 31
        assert number_of_days(2000, 7) == 31
        assert number_of_days(1998, 8) == 31
        assert number_of_days(2023, 9) == 30
    
>       assert number_of_days(1900, 2) == 28
E       assert 29 == 28
E        +  where 29 = number_of_days(1900, 2)

Calendar_CLI/tests/test_backed.py:52: AssertionError
```

---

This is the leap year logic as a reference.

```python
is_leap = (year % 400 == 0) or ((year % 4 ==0) and (year % 100 != 0)) 
```

File: Calendar_CLI/Calendar.py
Comment (YLee2121) [inline Calendar_CLI/Calendar.py:1]: ln 23, ln 28

the condition case is wrong.

Comment (viettran15) [reply Calendar_CLI/Calendar.py:1]: This bug should be fixed now. Thanks!

**Review by YLee2121 — APPROVED** [review]
Approve

**Pull request merged**